src/test/java/com/limechain/sync/fullsync/BlockExecutorTest.java

```diff
@@ -95,18 +75,43 @@ void kusamaFirstBlock() throws IOException {
         // Parse an expected babeslot error (expected since older runtimes since think this inherent will be passed to them, but it's deprecated)
         boolean checkOk = checkInherentsResponse[0] == 1;
         boolean fatalErrorEncountered = checkInherentsResponse[1] == 1;
-        var data = ScaleUtils.Decode.decode(ArrayUtils.subarray(checkInherentsResponse, 2, checkInherentsResponse.length),
-            new ListReader<>(new PairReader<>(scr -> new String(scr.readByteArray(8)), scr -> new String(scr.readByteArray()))));
+        var data =
+            ScaleUtils.Decode.decode(ArrayUtils.subarray(checkInherentsResponse, 2, checkInherentsResponse.length),
+                new ListReader<>(
+                    new PairReader<>(scr -> new String(scr.readByteArray(8)), scr -> new String(scr.readByteArray()))));
 
-        // Assert that the first kusama block is properly encoded as a parameter to the runtime
+        // Assert that the first Kusama block is properly encoded as a parameter to the runtime
         assertArrayEquals(
             StringUtils.hexToBytes("b0a8d493285c2df73290dfb7e61f870f17b41801197a149ca93654499ea3dafe04fabb0c6e92d29e8bb2167f3c6fb0ddeb956a4278a3cf853661af74a076fc9cb7a35fb7f7616f5c979d48222b3d2fa7cb2331ef73954726714d91ca945cc34fd8080642414245340201000000ef55a50f00000000044241424549040118ca239392960473fe1bc65f94ee27d890a49c1b200c006ff5dcc525330ecc16770100000000000000b46f01874ce7abbb5220e8fd89bede0adad14c73039d91e28e881823433e723f0100000000000000d684d9176d6eb69887540c9a89fa6097adea82fc4b0ff26d1062b488f352e179010000000000000068195a71bdde49117a616424bdc60a1733e96acb1da5aeab5d268cf2a572e94101000000000000001a0575ef4ae24bdfd31f4cb5bd61239ae67c12d4e64ae51ac756044aa6ad8200010000000000000018168f2aad0081a25728961ee00627cfe35e39833c805016632bf7c14da580090100000000000000000000000000000000000000000000000000000000000000000000000000000008280402000b90110eb36e011004140000"),
             executeBlockParam);
 
         // Attempt to execute a block
-        log.info("Executing block");
-        byte[] executeBlockResponse = genesisRuntime.call("Core_execute_block", executeBlockParam);
+        assertDoesNotThrow(() -> genesisRuntime.call("Core_execute_block", executeBlockParam),
+            "Executing the first block of Kusama must not throw any exceptions.");
+
+        // Explicitly assert that we calculated the expected state root
+        // TODO: This recalculates the merkle root again... where do we fetch the actual result from... the db... and how?
+        byte[] actualRoot = trieAccessor.getMerkleRoot(StateVersion.V0);
```

Comment:
@ablax That's the issue I was asking you about. How would you do this; what's your take on that?

Formally described here [#439](https://github.com/LimeChain/Fruzhin/issues/439)
Touches on things I considered out of scope for now, so it's up for debate.

Reply:
The assert on line 89 pretty much asserts that the expectedRoot is matching the actualRoot. Otherwise we do not purposefully store our own calculated merkleRoot anywhere. Maybe you can make the mainTrieRoot variable in TrieAccessor gettable, as it is updated on each getMerkleRoot call, and use it as the actualRoot.

Reply:
I would restrain myself from further refactoring `BlockTrieAccessor` here. If you really think we wouldn't need the actual computed root after a block to be accessible for us (as it's only ever a concern of the runtime), then I'm ok with removing the assertion completely. 

This, however, touches on another decision within `TrieAccessor` - it mutates itself on calling `getMerkleRoot`, becoming an accessor for the next block automatically. I really don't like this happening implicitly (at least because `getMerkleRoot` does a lot more than it says as a method), so I'd at least make this mutation explicit... for this though, some (at least internal) knowledge of what the next state root is would have to be persisted somewhere (either in the block's header, or as a variable on user-site).

Anyhow, that's a topic for a further PR.

Reply:
Delegated to the follow-up PR.

Reply:
After encapsulating things around block execution more properly, I felt like this test doesn't need to be so explicit about argument serialization and trie root assertions, so I dropped those.

Serializing the arguments for runtime invocations is now a concern of `Runtime`, and a private implementation detail (for now), hence generally untested. We'll test on a higher level for this one.

See e222d870a46af for the relevant changes.